### PR TITLE
Fix wrong code coverage for return statements in master

### LINKED ISSUE #32411: Invalid coverage for return statement when there's a if condition
**Description:**
For the following sample code:

`main.bal`

```
public function intAddd(int a, int b) returns int {
    if (true) {
        return a + b;
    } 
    return a - b;
}
```
`tests/main_test.bal`

```
import ballerina/test;

@test:Config{}
function testFunc() {
    test:assertEquals(intAddd(1,2), 3);
}
```

Execute bal test --code-coverage

Even though the final return  is never hit by the test, this code gives 100% coverage in Testerina report.

**Steps to reproduce:**
As discussed above

**Affected Versions:**
Swan lake beta 3

**OS, DB, other environment details and versions:**

**Related Issues (optional):**
The issue https://github.com/ballerina-platform/ballerina-lang/issues/31061 has been fixed, but the current issue is still there.

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject
Fixes #32411 

## Approach
This is due to jacoco takes line number for the byte codes coming after last line in the line number table as the same line. Fixing it by providing the line number of  `}` in the function for the injected bytecodes.

## Samples
```ballerina
public function intAdd(int a, int b, int val) returns int {
    if (val > 2) {
        return a + b;
    }
    return a - b;
}
```
in `test.bal`
```ballerina
import ballerina/test;

@test:Config {}
function testFunc() {
    test:assertEquals(intAdd(1, 2, 22), 3);
}
```
## Remarks
TODO: A fix need to be done to special case the step over functionality in the vscode debugger to ignore the `}` line number. @NipunaRanasinghe 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by HindujaB** [comment]
Code coverage before fix - 
before_fix

Code coverage after fix - 
after_fix

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java [hunk 1]
```diff
@@ -506,7 +506,14 @@ private void pushShort(MethodVisitor mv, int stateVarIndex, int caseIndex) {
 
     private void processTerminator(MethodVisitor mv, BIRFunction func, BIRPackage module, String funcName,
                                    BIRTerminator terminator) {
-        if (terminator.kind != InstructionKind.RETURN) {
+        if (terminator.kind == InstructionKind.GOTO &&
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java:509]: Shall we add a comment here

**Comment by HindujaB** [comment]
This fix is sent to the update branch https://github.com/ballerina-platform/ballerina-lang/pull/35082.
Hence, closing this PR.

**Comment by HindujaB** [comment]
Re-opening as this fix is needed for master as well.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**